tilelang/language/tir/ir.py

```
@@ -152,33 +162,318 @@ def grid(*extents: PrimExpr) -> frame.ForFrame:
         if "dtype" in kwargs:
             kwargs.pop("dtype")
         return func(*args, **kwargs)
 
     return wrapped
 
 
+if TYPE_CHECKING:
+    # Static signatures for the runtime bindings below. This replaces the
+    # former partial .pyi without adding another runtime wrapper layer.
+    # fmt: off
+    def abs(x: _T, span: Span | None = None) -> _T: ...
```

Ruff (0.16.0)

[error] Variable `abs` is shadowing a Python builtin

(A001)